src/main/scala/fi/oph/koski/json/SensitiveDataFilter.scala

```diff
@@ -23,15 +25,15 @@ case class SensitiveDataFilter(user: SensitiveDataAllowed) {
   }
 
   def sensitiveHidden(metadata: List[Metadata]): Boolean = metadata.exists {
-    case SensitiveData() => !user.sensitiveDataAllowed
+    case SensitiveData(requiredRoles) => !user.sensitiveDataAllowed(requiredRoles)
     case _ => false
   }
 }
 
 trait SensitiveDataAllowed {
-  def sensitiveDataAllowed: Boolean
+  def sensitiveDataAllowed(requiredRoles: Set[Role]): Boolean
```

Comment:
Ehkä nitpick, mutta jos käsitin tän oikein, niin pitäisikö "requiredRoles" olla "allowedRoles" tai "permittedRoles". 

requiredRoles kuulostaa siltä, että käyttäjällä pitäisi olla kaikki mainitut roolit ("AND"-tyyppinen).